docs/src/Groups/groups.md

```diff
@@ -22,6 +22,424 @@ Julia supports the following types of groups:
+```@docs
+abelian_group(::Type{T}, v::Vector{Int}) where T <: GAPGroup
+cyclic_group
+dihedral_group
+quaternion_group
+```
+!!! warning
+    The type need to be specified in the input of the function `abelian_group`, otherwise a group of type `GrpAbFinGen` is returned, which is not a GAP group type. In future versions of Oscar, this may change.
+
+The generators of a polycyclic group are displayed as `f1`, `f2`, `f3`, etc., and every element of a polycyclic group is displayed as product of such generators.
+
+  **Example:**
+```jldoctest
+julia> G=abelian_group(PcGroup, [2,4]);
+julia> G[1], G[2]
+(f1, f2)
+julia> G[2]*G[1]
+f1*f2
+```
+
+Note that this does not define Julia variables named `f1`, `f2`, etc.! To get the generators of the group `G`, use `gens(G)`; for convenience they can also be accessed as `G[1]`, `G[2]`, as shown in Section [Elements of groups](@ref elements_of_groups).
+
+## Homomorphisms
+
+In Oscar, a group homomorphism from `G` to `H` is an object of parametric type `GAPGroupHomomorphism{S,T}`, where `S` and `T` are the types of `G` and `H` respectively.
+
+A homomorphism from `G` to `H` can be defined in two ways.
+
+* Writing explicitly the images of the generators of `G`:
+```julia
+f = hom(G,H,[x1,x2,...],[y1,y2,...])
+```
+Here, `[x1,x2,...]` must be a generating set for `G` (not necessarily minimal) and `[y1,y2,...]` is a vector of elements of `H` of the same length of `[x1,x2,...]`. This assigns to `f` the value of the group homomorphism sending `x_i` into `y_i`. If such a homomorphism does not exist, an error is returned.
+
+* Taking an existing function `g` satisfying the group homomorphism properties:
+```julia
+f = hom(G,H,g)
+```
+If the function `g` does not satisfy the group homomorphism properties, an error is returned.
+
+  **Example:**
+The following procedures define the same homomorphism (conjugation by `x`) in the two ways explained above.
+```jldoctest
+julia> S=symmetric_group(4);
+julia> x=S[1];
+julia> f=hom(S,S,gens(S),[S[1]^x,S[2]^x]);
+julia> g=hom(S,S,y->y^x);
+julia> f==g
+true
+```
+
+Oscar has also the following standard homomorphism.
+```@docs
+id_hom
+trivial_morphism
+```
+
+To evaluate the homomorphism `f` in the element `x` of `G`, it is possible to use the instruction
+```julia
+image(f,x)
+```
+or the more compact notations `f(x)` and `x^f`.
+
+  **Example:**
+```jldoctest
+julia> S=symmetric_group(4);
+julia> f=hom(S,S,x->x^S[1]);
+julia> x=cperm(S,[1,2]);
+julia> image(f,x)
+(2,3)
+julia> f(x)
+(2,3)
+julia> x^f
+(2,3)
+```
+
+A sort of "inverse" of the evaluation is the following
+```@docs
+haspreimage(f::GAPGroupHomomorphism, x::GAPGroupElem)
+```
+  **Example:**
+```jldoctest
+julia> S=symmetric_group(4);
+julia> f=hom(S,S,x->x^S[1]);
+julia> x=cperm(S,[1,2]);
+julia> haspreimage(f,x)
+(true,(1,4))
+```
+
+!!! warning
+    Do not confuse `haspreimage` with the function `has_preimage`, which works on variable of type `GrpGenToGrpGenMor`.
```

Comment:
Couldn't these two functions be merged? It looks indeed confusing to have both in the same system.

Reply:
Two functions called `haspreimage` and `has_preimage` already exist in Hecke. The first one is defined on variables of types `GrpAbFinGenMap` and `GrpAbFinGenElem`, while the second is defined on variables of type `GrpGenToGrpGenMor` and `GrpGenElem`, and they do essentially the same thing. If we want to merge these two functions, this has to be done in the Hecke package first. Then, in Oscar, we can simply import the function we decide to preserve and extend it to `GAPGroups`.

Reply:
This is just an oversight/inconsistency in Hecke. We should not replicate this here :)

We just need to decide for one convention and follow it through. There is no need to change it in Hecke first.